src/app/shared/services/chat/chat.service.ts

```diff
@@ -1,51 +1,61 @@
 import { HttpClient, HttpParams } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Store } from '@ngxs/store';
 import { Observable } from 'rxjs';
+
 import { Role } from '../../enum/role';
 import { ChatRoom, ChatRoomsParameters, IncomingMessage, MessagesParameters } from '../../models/chat.model';
-import { PaginationElement } from '../../models/paginationElement.model';
 import { SearchResponse } from '../../models/search.model';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ChatService {
-  constructor(private http: HttpClient, private store: Store) {}
+  constructor(private http: HttpClient) {}
 
   getChatRooms(parameters: ChatRoomsParameters): Observable<SearchResponse<ChatRoom[]>> {
-    let params = this.setFilterParams(parameters);
+    const params = this.setChatRoomParams(parameters);
 
     return this.http.get<SearchResponse<ChatRoom[]>>(`/api/v1/ChatWorkshop/${parameters.role}/chatrooms`, { params });
   }
 
-  getChatRoomsMessages(chatRoomId: string, role: Role, parameters: MessagesParameters): Observable<IncomingMessage[]> {
-    let params = new HttpParams().set('Size', parameters.size.toString()).set('From', parameters.from.toString());
+  getChatRoomsForProviderByParentId(parentId: string): Observable<ChatRoom[]> {
+    return this.http.get<ChatRoom[]>(`/api/v1/ChatWorkshop/provider/chatroomsforparent/${parentId}`);
```

Comment:
`/api/v1/ChatWorkshop` this part can be a variable e.g. `baseApiUrl` or something similar